lib/portlayer/network/context.go

```diff
@@ -1073,10 +1086,20 @@ func (c *Context) AddContainer(h *exec.Handle, options *AddContainerOptions) err
 		h.ExecConfig.Networks = make(map[string]*executor.NetworkEndpoint)
 	}
 
+	// If the scope type is external, name the nic as such.  We can only have
```

Comment:
The `Network.Type` field in the `NetworkEndpoint` already has the type, so you can use that.